### PR TITLE
send timestamp metric to datadog

### PR DESCRIPTION
**Pull request by sravfeyn**

### CODE REVIEW

File: corehq/ex-submodules/pillowtop/pillow/interface.py [hunk 1]
```diff
@@ -206,7 +206,8 @@ def __record_change_metric_in_datadog(self, metric, change, timer=None):
                 u'document_type:{}'.format(change.metadata.document_type),
                 u'domain:{}'.format(change.metadata.domain),
                 u'is_deletion:{}'.format(change.metadata.is_deletion),
-                u'pillow_name:{}'.format(self.get_name())
+                u'pillow_name:{}'.format(self.get_name()),
+                u'publist_timestamp:{}'.format(change.metadata.publish_timestamp),
```
Comment (snopoke) [inline corehq/ex-submodules/pillowtop/pillow/interface.py]: This won't work - you need a new guage metric which records the timestamp as an epoch

**Review by snopoke — CHANGES_REQUESTED** [review]

**Comment by sravfeyn** [comment]
@snopoke Updated

File: corehq/ex-submodules/pillowtop/pillow/interface.py [hunk 2]
```diff
@@ -206,10 +207,13 @@ def __record_change_metric_in_datadog(self, metric, change, timer=None):
                 u'document_type:{}'.format(change.metadata.document_type),
                 u'domain:{}'.format(change.metadata.domain),
                 u'is_deletion:{}'.format(change.metadata.is_deletion),
-                u'pillow_name:{}'.format(self.get_name())
+                u'pillow_name:{}'.format(self.get_name()),
             ]
             datadog_counter(metric, tags=tags)
 
+            change_lag = (datetime.utcnow() - change.metadata.publish_timestamp).seconds
+            datadog_gauge('commcare.change_feed.change_lag', change_lag, tags=tags)
```
Comment (snopoke) [inline corehq/ex-submodules/pillowtop/pillow/interface.py]: I don't think all these tags make sense for this metric. I think we could limit it to:
* pillow_name
* topic

I think you may have to add the topic to the change though. You can add it here: https://github.com/dimagi/commcare-hq/blob/be5c8ce9ea6f9cf1c16894570505172911ccbb7c/corehq/apps/change_feed/consumer/feed.py#L179
```
topic = message.topic
```

**Review by snopoke — CHANGES_REQUESTED** [review]

File: corehq/ex-submodules/pillowtop/pillow/interface.py (hunk 2 above)
Comment (sravfeyn) [inline corehq/ex-submodules/pillowtop/pillow/interface.py]: https://github.com/dimagi/commcare-hq/pull/15544/commits/33f627431237443bdc2b4a45aa35f8d6925e2d67

File: corehq/apps/change_feed/consumer/feed.py [hunk 3]
```diff
@@ -181,6 +181,7 @@ def change_from_kafka_message(message):
         deleted=change_meta.is_deletion,
         metadata=change_meta,
         document_store=document_store,
+        topic=message.topic,
```
Comment (snopoke) [inline corehq/apps/change_feed/consumer/feed.py]: You'll need to add this kwarg to the `Change` class as well

**Review by snopoke — CHANGES_REQUESTED** [review]

File: corehq/apps/change_feed/consumer/feed.py (hunk 3 above)
Comment (sravfeyn) [inline corehq/apps/change_feed/consumer/feed.py]: Sorry, thought it was already there

Updated now

**Review by snopoke — APPROVED** [review]

**Comment by sravfeyn** [comment]
Is there anything else needed to be done on datadog UI for this metric to reflect there?

**Comment by snopoke** [comment]
@sravfeyn you'll need to add a graph to see the data but you don't have to do anything for it to be collected

**Pull request merged**